### DISCUSSION TITLE
Using PlutoTest in Nbdev.jl

### DISCUSSION THREAD

**Discussion by sapal6** [JuliaPluto/PlutoTest.jl#21] [Q&A] [answered]
Hi There. Can someone let me know if at this stage (alpha) PlutoTest is ready to be implemented in my project?

I am developing a port of fastai's Nbdev(python) into [Julia](https://sapal6.github.io/nbdev.jl). I was trying to think about ways to enable users to carry out unit tests inside Pluto notebooks and then enable execution of tests using github actions. PlutoTest looks like a good match for my requirements and I would like to try it out in my project.

**Comment by fonsp** [comment, marked as answer]
Hi! We believe that we have already created a port of nbdev: it's Pluto! For example, these packages are written and tested using Pluto notebooks:
- https://github.com/JuliaPluto/PlutoTest.jl
- https://github.com/JuliaPluto/PlutoUI.jl
- https://github.com/JuliaPluto/PlutoHooks.jl
- https://github.com/JuliaPluto/PlutoLinks.jl

**Reply by sapal6** [reply]
Hey  @fonsp thanks for the reply. My project Nbdev.jl is itself based on pluto. Using Nbdev.jl one can start off with pluto notebook as a starting point and generate documentation (hosted on github pages) and code files (source code only) .  I was trying to build a module to enable users to write tests inside Pluto but came across PlutoTests.jl which I think can be integrated in Nbdev.jl . Only missing part here is executing the tests with github actions which I read in PlutoTests's Readme as one of the "coming soon" features.

**Reply by fonsp** [reply]
Hi @sapal6 ! 

Take a look at the packages I linked above, open the source code in pluto to learn the mechanisms that we have for `export`, `hide`, etc. My point is that we don't need a generation step, this is necessary in python/jupyter (hence nbdev), but not with Pluto.

Also join the pluto developers call next Thursday at 18:00 CET! We will probably be discussing this topic. Take a look at the [pluto developers zulip](https://gist.github.com/fonsp/db7d00fd3fe5bc0b379b4af9ec6674b6) for info.

**Reply by sapal6** [reply]
Hi @fonsp. I always appreciate your quick reply. I agree to the points. However, the need for nbdev.jl arose due to the fact that in some situations of work there was a need to have separate Julia source files with the required source code only and separate document files in the form of markdown. 

The idea is to have an optional package to have an ability to do the above things in addition to what is already possible in native Pluto like hosting source files as statuc html, executing Pluto notebooks as native julia source ( since Pluto notebooks are Julia files).

However, I believe that Nbdev.jl had always benefited from the out-of-box functionalities available in Pluto and looking at the source, have made it possible for me to improve the code in Nbdev.

The call timings are a bit late from India time perspective but I will keep an eye on the Zulip channel.

**Reply by fonsp** [reply]
Sorry to hear that the time does not work for you :( it's impossible to find a time that works for everyone.

Since most pluto developers live in europe, we can set up a call earlier in the day, would that work for you? Send me a message on zulip to find a time, might be easier than github discussions

**Reply by sapal6** [reply]
Thanks for considering. Actually, my nature of work makes it difficult to make out time for morning and evening calls. However, I will try to make time for the call whenever I can but I will keep an eye on the zulip discussion.
